test/extended/util/disruption/frontends/known_backends.go

```diff
@@ -4,55 +4,109 @@ import (
 	"context"
 
 	configclient "github.com/openshift/client-go/config/clientset/versioned"
+	routeclient "github.com/openshift/client-go/route/clientset/versioned"
 	"github.com/openshift/origin/pkg/monitor"
 	"github.com/openshift/origin/pkg/monitor/backenddisruption"
 	"github.com/openshift/origin/test/extended/util/cluster"
+	apierrors "k8s.io/apimachinery/pkg/api/errors"
 	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
 	utilruntime "k8s.io/apimachinery/pkg/util/runtime"
+	"k8s.io/client-go/discovery"
 	"k8s.io/client-go/rest"
 	e2e "k8s.io/kubernetes/test/e2e/framework"
 )
 
+const (
+	oauthRouteNamespace = "openshift-authentication"
+	oauthRouteName      = "oauth-openshift"
+)
+
 func StartAllIngressMonitoring(ctx context.Context, m monitor.Recorder, clusterConfig *rest.Config) error {
-	if err := createOAuthRouteAvailableWithNewConnections().StartEndpointMonitoring(ctx, m, nil); err != nil {
+	// Ingress monitoring checks for oauth and console routes to monitor healthz endpoints. Check availability
+	// before setting up any monitors.
+	available, err := isRouteAvailable(ctx, clusterConfig, oauthRouteNamespace, oauthRouteName)
+	if err != nil {
 		return err
 	}
-	if err := createOAuthRouteAvailableWithConnectionReuse().StartEndpointMonitoring(ctx, m, nil); err != nil {
-		return err
+	if available {
+		if err := createOAuthRouteAvailableWithNewConnections().StartEndpointMonitoring(ctx, m, nil); err != nil {
+			return err
+		}
+		if err := createOAuthRouteAvailableWithConnectionReuse().StartEndpointMonitoring(ctx, m, nil); err != nil {
+			return err
+		}
 	}
 
-	// Some jobs explicitly disable the console and other features. Check if it's disabled and if so,
-	// do not run a disruption monitoring backend for it.
-	configClient, err := configclient.NewForConfig(clusterConfig)
+	configAvailable, err := isConfigAPIGroupAvailable(clusterConfig)
 	if err != nil {
 		return err
 	}
-	clusterVersion, err := configClient.ConfigV1().ClusterVersions().Get(context.TODO(), "version", metav1.GetOptions{})
-	if err != nil {
-		e2e.Failf("Failed to get cluster version: %v", err)
-	}
-	// If the cluster does not know about the Console capability, it likely predates 4.12 and we can assume
-	// it has it by default. This is to catch possible future scenarios where we upgrade 4.11 no cap to 4.12 no cap.
-	if !cluster.KnowsCapability(clusterVersion, "Console") ||
-		cluster.HasCapability(clusterVersion, "Console") {
-		if err := createConsoleRouteAvailableWithNewConnections().StartEndpointMonitoring(ctx, m, nil); err != nil {
+	if configAvailable {
+		// Some jobs explicitly disable the console and other features. Check if it's disabled and if so,
+		// do not run a disruption monitoring backend for it.
+		configClient, err := configclient.NewForConfig(clusterConfig)
```

Comment:
If `exutil.DoesApiResourceExist` gets used, the oc's config client could be used here instead of creating a new one.

Reply:
This would involve creating an oc object here only to use the config client, is that an overkill? oc objects do many operations besides creating clients for APIs.

Reply:
If that's the case I am taking my comment back